### PR TITLE
Add SQLite store

### PR DESCRIPTION
**Pull request by n8maninger**

### CODE REVIEW

File: persist/sqlite/types.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+type decodableSlice[T any] struct {
+	v *[]T
+	n int64
+}
+
+func (d *decodableSlice[T]) Scan(src any) error {
+	switch src := src.(type) {
+	case []byte:
+		dec := types.NewDecoder(io.LimitedReader{
```
Comment (lukechampine) [inline persist/sqlite/types.go:87]: `types.NewBufDecodr`

File: api/client.go [hunk 2]
```diff
@@ -105,20 +105,20 @@ func (c *Client) Wallet(name string) *WalletClient {
 	return &WalletClient{c: c.c, name: name}
 }
 
+// Resubscribe subscribes the wallet to consensus updates, starting at the
+// specified height. This can only be done once.
+func (c *Client) Resubscribe(height uint64) (err error) {
```
Comment (lukechampine) [inline api/client.go:110]: only once?

File: cmd/walletd/testnet.go [hunk 3]
```diff
@@ -115,12 +115,13 @@ func initTestnetClient(addr string, network string, seed wallet.Seed) *api.Clien
 		} else if err := wc.AddAddress(ourAddr, nil); err != nil {
 			fmt.Println()
 			log.Fatal(err)
-		} else if err := wc.Subscribe(0); err != nil {
-			fmt.Println()
-			log.Fatal(err)
 		}
-		fmt.Println("done.")
 	}
+	if err := c.Resubscribe(0); err != nil {
```
Comment (lukechampine) [inline cmd/walletd/testnet.go:120]: won't this resubscribe every time `initTestnetClient` is called? That seems... undesirable.

File: persist/sqlite/sql.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+)
+
+type (
+	// A scanner is an interface that wraps the Scan method of sql.Rows and sql.Row
+	scanner interface {
+		Scan(dest ...any) error
+	}
+
+	// A txn is an interface for executing queries within a transaction.
+	txn interface {
```
Comment (lukechampine) [inline persist/sqlite/sql.go:26]: what are these interfaces used for?

File: wallet/wallet.go [hunk 5]
```diff
@@ -143,12 +150,17 @@ type Event struct {
 	Index     types.ChainIndex
 	Timestamp time.Time
 	Relevant  []types.Address
-	Val       interface{ eventType() string }
+	Val       interface{ EventType() string }
```
Comment (lukechampine) [inline wallet/wallet.go:153]: I'm not seeing this used anywhere

File: persist/sqlite/sql.go (hunk 4 above)
Comment (n8maninger) [reply persist/sqlite/sql.go:26]: It hides the commit and rollback functions from the *sql.Tx to enforce transaction handling.

File: cmd/walletd/testnet.go (hunk 3 above)
Comment (n8maninger) [reply cmd/walletd/testnet.go:120]: Huh. Yeah... Not sure why I moved that.

File: persist/sqlite/sql.go (hunk 4 above)
Comment (lukechampine) [reply persist/sqlite/sql.go:26]: ok, `scanner` makes sense to me. `txn` looks like it's only being implemented by `loggedTxn` (?) so might not be strictly necessary.

**Pull request merged**